common-requirements.md

```diff
@@ -0,0 +1,30 @@
+## Title
+Common Requirements
+
+## Context
+Many projects are trying to use common code.  There is a shared repository that all the projects access.  This pattern applies if there is a Strong Code Owner [pattern to be written] or if there is weak code ownership, or no Benevolent Sponsor [pattern to be written]. Someone (or some project) wrote the code in the first place and contributed it to the repository. The common code is a small percentage of the overall deliverable from any of the projects.  Each project has its own delivery schedule, set of deliverables and customers.
+
+## Problem Statement
+The common code in the shared repository isn't meeting the needs of all the projects that want to use it.  
+
+## Forces
+The project that made the code available has one set of needs. Its needs are similar to what some of the receiving organization wants, but not quite the same.
+Requirements on code should be derivable from real customer needs.  
+The needs of different customers are generally quite similar; however they might be expressed differently or weighted differently between customers.
+Many customers want the supplier to help them know what they need.
+The company has many “Systems Engineers” writing requirements for the products.  These requirements are supposed to be a distillation of customer needs to guide development of the product. 
+Reusing code is an important goal to save the company time and money.  
+
+## Solution
+Align the requirements of the projects so that the code that meets the requirements for one project also meets the needs for the other projects.  Decompose the code into smaller pieces for which the many using projects can agree upon requirements.
+
+Additionally, take advantage of customers expecting the supplier to help elucidate requirements. Bring about the alignment of requirements during the customer negotiations and influence the customers requirements rather than changing the component.
+
```

Comment:
From my old product line days I remember, that bringing about such an alignment requires an incentive for both the customers (usually price and TTM) _and_ for the company itself. Should we mention this here or would that unduly bloat the pattern? Maybe this could go in the _Resulting Context_ section, too.

Reply:
This sounds like a reasonable thing to add to the pattern at this location.

Reply:
That's a good comment for resulting context.  It's a reflection that the problem (getting alignment) might introduce new problems (that require solutions like discounts).